src/main/java/spoon/support/compiler/VirtualFile.java

```diff
@@ -44,6 +46,17 @@ public InputStream getContent() {
 		return new ByteArrayInputStream(content.getBytes());
 	}
 
+	@Override
+	public char[] getContentChars(Environment env) {
+		byte[] bytes = content.getBytes();
+		if (env.getEncodingProvider() == null) {
+			return new String(bytes, env.getEncoding()).toCharArray();
+		} else {
+			Charset encoding = env.getEncodingProvider().detectEncoding(this, bytes);
+			return new String(bytes, encoding).toCharArray();
+		}
+	}
```

Comment:
`content.getBytes()` was an old bug in Spoon, E.g. if content contains special characters and then you use getBytes, which encodes bytes using system (potentially wrong) encoding, then information is lost.
The correct implementation is simply
```java
char[] getContentChars(Environment env) {
   return content.toCharArray();
}
```
because it makes no sense to ask for `byte array encoding`, if we already have `char array` stored in the `String content`.